File: Travel Package Management Java/src/com/example/travel/GoldPassenger.java
package com.example.travel;

public class GoldPassenger extends Passenger {
    public GoldPassenger(String name, int passengerNumber, double balance) {
        super(name, passengerNumber, balance);
    }

    @Override
    public boolean signUpForActivity(Destination destination, Activity activity) {
        double cost = calculateActivityCost(activity);
        if (getBalance() >= cost && !activity.isSignedUp(this)) {
            if (activity.signUp(this)) {
                deductBalance(cost);
                return true;
            }
        }
        return false;
    }

    @Override
    public double calculateActivityCost(Activity activity) {
        return activity.getCost() * 0.9;
    }
}

File: Travel Package management C++/GoldPassenger.cpp
#include "GoldPassenger.h"


using namespace std;


bool GoldPassenger::signUpForActivity(shared_ptr<Destination> destination, shared_ptr<Activity> activity){
    double cost = activity->getCost() * 0.9;
    if(getBalance() >= cost && !activity->isSignedUp(shared_from_this())){
        if(activity->signUp(shared_from_this())){
            deductBalance(cost);
            return true;
        }
    }
    return false;
} 

double GoldPassenger::calculateActivityCost(shared_ptr<Activity> activity){
    return activity->getCost() * 0.9; 
}   